src/providers.js

```diff
@@ -219,67 +176,55 @@ class Providers {
   }
 
   /**
-   * Add a new provider.
+   * Add a new provider for the given CID.
    *
    * @param {CID} cid
    * @param {PeerId} provider
-   * @param {function(Error)} callback
-   * @returns {undefined}
+   * @returns {Promise}
    */
-  addProvider (cid, provider, callback) {
-    this._log('addProvider %s', cid.toBaseEncodedString())
-    const dsKey = makeProviderKey(cid)
-    const provs = this.providers.get(dsKey)
-
-    const next = (err, provs) => {
-      if (err) {
-        return callback(err)
-      }
+  async addProvider (cid, provider) {
```

Comment:
```suggestion
  addProvider (cid, provider) {
```

Reply:
here's what I wrote in #134 :
> I don't know if there are any significant perf impacts, but i find this more readable

> when i see async i can clearly tell it returns a promise
> when i see await instead of a return i can clearly see it doesn't return any value